lib/fluent/plugin/filter_process_ucs_syslog.rb

```diff
@@ -46,7 +53,12 @@ def filter(tag, time, record)
       record["device"] = ""
       record["error"] = ""
 
-      fullUsername = "#{domain}\\#{username}"
+      if @domain.nil?
+        fullUsername = "#{username}"
+      else
+        fullUsername = "#{domain}\\#{username}"
+      end
+      
```

Comment:
This code is used at least twice. I think it's better to extract it to a single function and reuse it.

Reply:
added.